apex/optimizers/fused_mixed_precision_lamb.py

```diff
@@ -0,0 +1,250 @@
+import torch
+from copy import deepcopy
+from itertools import chain
+from collections import defaultdict, abc as container_abcs
+
+from apex.multi_tensor_apply import multi_tensor_applier
+
+class FusedMixedPrecisionLamb(torch.optim.Optimizer):
+
+    def __init__(self, params, lr=1e-3, step=0, bias_correction=True,
+                 betas=(0.9, 0.999), eps=1e-6, weight_decay=0.01,
+                 amsgrad=False, adam_w_mode=True,
+                 grad_averaging=True, max_grad_norm=1.0, use_nvlamb=False,
+                 reduced_precision_dtype=None):
+        if amsgrad:
+            raise RuntimeError('FusedLAMB does not support the AMSGrad variant.')
+        
+        device = params[0].device
+
+        # The learning rate (lr) and optimizer step (step) should be located on device
+        # in order to faciliated device sync free execution
+        defaults = dict(lr=torch.tensor(lr, dtype=torch.float32, device=device),
+                        step=torch.tensor([step], dtype=torch.int, device=device),
```

Comment:
I'm sorry I completely missed that `params` can be a `dict`.
For example, https://github.com/NVIDIA/DeepLearningExamples/blob/fd9fecd2b22e6b9e25e75de8b0a90a711cf91477/PyTorch/LanguageModeling/BERT/run_pretraining.py#L376-L38484

So moving `lr` and `step` tensors after `super().__init__(params, defaults)` because we'll have `param_groups` attribute for sure.
```suggestion

        # The learning rate (lr) and optimizer step (step) should be located on device
        # in order to faciliated device sync free execution.
        # That being said, `params` is either `dict` or `list`
        # so we're sending these two tensors to device after `super().__init__()`.
        defaults = dict(lr=torch.tensor(lr, dtype=torch.float32),
                        step=torch.tensor([step], dtype=torch.int),
```